google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/apis/GoogleCloudApiSelectorPanelTest.java

```diff
@@ -322,6 +325,90 @@ public void getSelectedLibraries_withSomeSelected_returnsLibraries() {
+    GoogleCloudApiSelectorPanel panel =
+        new GoogleCloudApiSelectorPanel(ImmutableList.of(library), testFixture.getProject());
+    JTable table = panel.getCloudLibrariesTable();
+    checkCheckbox(table, 0);
+
+    Set<CloudLibrary> apisToEnable = panel.getApisToEnable();
+    assertThat(apisToEnable).containsExactly(library);
+  }
+
+  @Test
+  public void
+      getApisToEnable_withAllLibrariesChecked_AndNoShouldEnableChecked_returnsNoneEnabled() {
+    CloudLibrary library1 = LIBRARY_1.toCloudLibrary();
+    CloudLibrary library2 = LIBRARY_2.toCloudLibrary();
+
+    GoogleCloudApiSelectorPanel panel =
+        new GoogleCloudApiSelectorPanel(
+            ImmutableList.of(library1, library2), testFixture.getProject());
+    JTable table = panel.getCloudLibrariesTable();
+
+    checkCheckbox(table, 0);
+    checkCheckbox(table, 1);
+
+    Map<CloudLibrary, CloudApiManagementSpec> apiManagementMap = panel.getApiManagementMap();
+
+    panel.getDetailsPanel().setCloudLibrary(library1, apiManagementMap.get(library1));
+    panel.getDetailsPanel().getEnableApiCheckbox().setSelected(false);
+
+    panel.getDetailsPanel().setCloudLibrary(library2, apiManagementMap.get(library2));
+    panel.getDetailsPanel().getEnableApiCheckbox().setSelected(false);
+
+    assertThat(panel.getApisToEnable()).isEmpty();
+  }
+
+  @Test
+  public void
+      getApisToEnable_withAllLibrariesChecked_AndSomeShouldEnableChecked_returnsSomeEnabled() {
+    CloudLibrary library1 = LIBRARY_1.toCloudLibrary();
+    CloudLibrary library2 = LIBRARY_2.toCloudLibrary();
+
+    GoogleCloudApiSelectorPanel panel =
+        new GoogleCloudApiSelectorPanel(
+            ImmutableList.of(library1, library2), testFixture.getProject());
+    JTable table = panel.getCloudLibrariesTable();
+
+    checkCheckbox(table, 0);
+    checkCheckbox(table, 1);
+
+    Map<CloudLibrary, CloudApiManagementSpec> apiManagementMap = panel.getApiManagementMap();
+
+    panel.getDetailsPanel().setCloudLibrary(library1, apiManagementMap.get(library1));
+    panel.getDetailsPanel().getEnableApiCheckbox().setSelected(false);
+
+    panel.getDetailsPanel().setCloudLibrary(library2, apiManagementMap.get(library2));
+    panel.getDetailsPanel().getEnableApiCheckbox().setSelected(true);
+
+    assertThat(panel.getApisToEnable()).containsExactly(library2);
+  }
+
+  @Test
+  public void
+      getApisToEnable_withNoLibrariesChecked_AndAllShouldEnableChecked_returnsNoneEnabled() {
+    CloudLibrary library1 = LIBRARY_1.toCloudLibrary();
+    CloudLibrary library2 = LIBRARY_2.toCloudLibrary();
+
+    GoogleCloudApiSelectorPanel panel =
+        new GoogleCloudApiSelectorPanel(
+            ImmutableList.of(library1, library2), testFixture.getProject());
+
+    Map<CloudLibrary, CloudApiManagementSpec> apiManagementMap = panel.getApiManagementMap();
+
+    panel.getDetailsPanel().setCloudLibrary(library1, apiManagementMap.get(library1));
+    panel.getDetailsPanel().getEnableApiCheckbox().setSelected(true);
```

Comment:
BTW, regarding this test case, how about disabling that checkbox at all if library is not enabled? Otherwise a user might get a feeling he is able to enable Cloud API on GCP side without adding libraries to the module and project - the checkbox is still available.

Reply:
good point. thank you for reminding me of this.

Reply:
updated for this behavior: https://github.com/GoogleCloudPlatform/google-cloud-intellij/pull/1802/commits/42898ba73c9eabfab7c92ead3734e8dcc3bf04fe

Reply:
If we updated checkboxes to be disabled, do we still need to set them selected here? Could we also test they are disabled in this case?

Reply:
ah thanks. I forgot to update these tests after adding the disabled feature

Reply:
is it truly redundant to set the checkbox to selected even if its disabled? This test is still testing that panel returns no APIs to enable when the checkboxes are disabled+checked.

Reply:
(I agree that I still need to add assertions to confirm the enabled/disabled states)

Reply:
Agree, selection makes tests more thorough